emr_launcher.py

```diff
@@ -18,28 +20,31 @@ def configure_log():
     logger = logging.getLogger("emr_launcher")
     logger.propagate = False
     console_handler = logging.StreamHandler()
-    formatter = jsonlogger.JsonFormatter(
-        "%(asctime)s %(name)-12s %(levelname)-8s %(message)s"
-    )
-    console_handler.setFormatter(formatter)
+    #formatter = jsonlogger.JsonFormatter(
+    #    "%(asctime)s %(name)-12s %(levelname)-8s %(message)s"
+    #)
+    #console_handler.setFormatter(formatter)
     logger.addHandler(console_handler)
     return logger
 
 
-def read_s3_config(bucket: str, key: str, required: bool = True) -> dict:
+def read_s3_config(bucket: str, key: str, secrets: dict, required: bool = True) -> dict:
     config = {}
     s3_client = boto3.client("s3")
     try:
         response = s3_client.get_object(Bucket=bucket, Key=key)
-        config = yaml.safe_load(response["Body"])
+        with open(key, "w") as f:
```

Comment:
Why do we write the contents of the response body, just to read it back in again?

If it's just so we can do secrets replacement, we could probably change `replace_values()` to work directly on the values in the dict than read and write files.

Reply:
Fixed by https://github.com/dwp/emr-launcher/pull/2/commits/33b22f9189910869c8af1d7f433ba0e673827efd